### PR TITLE
Tickets/dm 41740

### PR DESCRIPTION
**Pull request by pcortes80**
- Create the `queueStateMachine` in **tcpServerBase** class.
- Put creation of `queueTcpServer`, `queueTcpClient`, `queueStateMachine`, and `queueTcpClientFromDds` in **FactoryTcpServer.createTcpServer.vi**.
- Remove **runServer.vi** and **tcpServerStateMachine.vi** from **TcpServerBase** class.
- Set **runServer.vi** and **tcpServerStateMachine.vi** as `preallocated clones` in **tcpServerCmd** and **tcpServerTel** classes.
- Release queue references using the **TcpServerBase.releaseQueues.vi**.
This vi is put into the **TcpServerBase.destroyReferences.vi**.
- Update unit tests to open and close the queue references properly.

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -111,13 +111,21 @@ Note:
 1. The polymorphic `convertToKeyValuePair.vi` does not support `Inf` value.
 Please try to avoid use this value.
 
+### List of Reentrant VI in TcpServer
```
Comment (teweitsai) [inline src/README.md:114]: Maybe the "preallocated clones" vi is more clear?

**Comment by teweitsai** [comment]
[Test]

<startTcpServerA.vi>

1. Do you still want to keep the "Diagram Disable Structure"?

<startTcpServerB.vi>

1. Do you still want to keep the "Diagram Disable Structure"?

[TcpServerBase]

<Write_queueStateMachine.vi>

1. In the doc string, "Write the queueStateMachine. This queue was created to change states in the state machine inside of the tcpServerStateMachine.vi."

**Review by teweitsai — APPROVED** [review]
Looks good!

File: src/README.md (hunk 1 above)
Comment (pcortes80) [reply src/README.md:114]: Done.

**Pull request merged**